metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -859,21 +857,18 @@ def get_profile_if_enabled(
                     if self.source_config.profiling.partition_patterns.allowed(
                         partition_spec
                     ):
-                        mcps.append(
-                            self._create_profile_mcp(
-                                mce, table_stats, column_stats, partition_spec
-                            )
-                        )
+                        yield self._create_profile_mcp(
+                            mce, table_stats, column_stats, partition_spec
+                        ).as_workunit()
```

Comment:
we could use the `auto_workunit` helper to remove the as_workunit calls too

Reply:
It'd be nice to emit mcps rather than workunits, but I'd rather not put `Union[MetadataWorkUnit, MetadataChangeProposalWrapper]` all over the place. Until we can emit only mcps would prefer to just call `.as_workunit()`. Open to discussion tho